### ISSUE TITLE
Arduino mega does not break out enough input capture pins

### ISSUE THREAD

**Issue by ihartwig** [CMU-Robotics-Club/RoboBuggy#123]
We can only get to ICP4 and ICP5, but we need to read 3 RC signals. I think we should do this with either standard interrupt pins (what we're doing now) or a pin change interrupt mask.

**Comment by ramensandwich** [comment]
What are the three signals which we'll be reading?

**Comment by ihartwig** [comment]
I noticed this while working on the hardware usage map (https://github.com/CMU-Robotics-Club/RoboBuggy/wiki/Hardware-Usage-Mega). That might help make this more clear. We are sending out 1 servo pulse (steering servo) and reading 3 of them (steering, brake, auton state) from the RC receiver. Each 16-bit timer can send 2 or 3 in hardware depending on the number of output compares it has, but it can only read 1 signal on the 1 input capture pin. Unfortunately, the arduino does not connect ICP1 or ICP3 to a pin header.

We could add a header and solder directly to the pin. I think we can make this robust enough, but I'm not sure it is worth the effort.

**Comment by TrevorDecker** [comment]
has this problem been solved?

**Comment by ihartwig** [comment]
It has been resolved-ish in that we wrote new rc receiver code based on
external interrupts rather than input capture. Based on watching the
interrupts on a scope and actual operation, this should work fine. We're
waiting on a full serial reading implementation before closing this.

On Sunday, April 5, 2015, Trevor Decker <notifications@github.com> wrote:

> has this problem been solved?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CMU-Robotics-Club/RoboBuggy/issues/123#issuecomment-89866635>
> .
>


-- 

Ian Hartwig [spacepenguine@gmail.com]
